alleycats-core/.jvm/src/main/scala/alleycats/SystemIdentityHash.scala

```diff
@@ -0,0 +1,13 @@
+package alleycats
+
+import cats.Hash
+
+class SystemIdentityHash[A <: AnyRef] extends ReferentialEq[A] with Hash[A] {
```

Comment:
Maybe just me, but perhaps make it a trait so that it can be mixed with other instances?

Reply:
I don't mind either way. I'll modify the change